independent-projects/resteasy-reactive/server/runtime/src/main/java/org/jboss/resteasy/reactive/server/core/StreamingUtil.java

```diff
@@ -58,9 +58,14 @@
         for (MessageBodyWriter<Object> writer : writers) {
             // Spec(API) says we should use class/type/mediaType but doesn't talk about annotations 
             if (writer.isWriteable(entityClass, entityType, Serialisers.NO_ANNOTATION, mediaType)) {
-                // FIXME: spec doesn't really say what headers we should use here
-                writer.writeTo(entity, entityClass, entityType, Serialisers.NO_ANNOTATION, mediaType,
-                        Serialisers.EMPTY_MULTI_MAP, baos);
+                // FIXME: Hack to enforce double quotes being written (our JSON writers won't do due to some TCK hack)
+                if (entityClass.equals(String.class) && mediaType.toString().startsWith(MediaType.APPLICATION_JSON)) {
+                    baos.write(("\"" + entity + "\"").getBytes(StandardCharsets.UTF_8));
```

Comment:
Shouldn't double quotes in `entity` be escaped somehow?

Reply:
Hmm, absolutely! We would also need to escape the other reserved characters, like linefeed, tab etc. I don't like the idea of putting all that escaping stuff into `StreamingUtil`. If you don't mind, I would rather try to fix the JSON writers, so that they won't extra-quote simple strings. WDYT?

Reply:
Go ahead and give it a shot